File: source/python/error_cleanup_defer.py
cleaned = False

def cleanup():
    global cleaned
    cleaned = True
    print("resource cleaned up")

def risky_operation():
    try:
        raise Exception("something went wrong")
    finally:
        cleanup()

def main():
    try:
        risky_operation()
    except Exception:
        print("error handled")

main()

File: source/go/error_cleanup_defer.go
package main

import "fmt"

var cleaned = false

func cleanup() {
	cleaned = true
	fmt.Println("resource cleaned up")
}

func riskyOperation() (err error) {
	_ = make([]byte, 64)
	defer cleanup()
	defer func() {
		// simulate free
	}()
	panic("something went wrong")
}

func main() {
	func() {
		defer func() {
			if r := recover(); r != nil {
				fmt.Println("error handled")
			}
		}()
		riskyOperation()
	}()
}
